src/main/java/com/johnstarich/moviematcher/models/User.java

```diff
@@ -0,0 +1,191 @@
+import com.mongodb.client.result.DeleteResult;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+import org.mindrot.jbcrypt.BCrypt;
+import static java.util.Arrays.asList;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static com.mongodb.client.model.Filters.eq;
+
+/**
+ * Created by Josue on 4/7/2016.
+ */
+public class User {
+    private static class LazyUsersCollection {
+        public static final MongoCollection<Document> usersCollection = MovieMatcherDatabase.getCollection("users");
+    }
+    private static MongoCollection<Document> getCollection() { return LazyUsersCollection.usersCollection; }
+
+    /* need to decide how to store user preferences ... */
+
+    public final ObjectId _id;
+    public final String email;
+    public final String first_name;
+    public final String last_name;
+    public final List<User> friends;
+    public final List<?> groups; // ? should be replaced with Group class
+    public final String password;
+
+    public User(ObjectId _id) {
+        this._id = _id;
+        this.email = null;
+        this.first_name = null;
+        this.last_name = null;
+        this.friends = new ArrayList<>(0);
+        this.groups = new ArrayList<>(0);
+        this.password = null;
+    }
+
+    public User(ObjectId _id, String email, String first_name, String last_name, List<User> friends, List groups, String password) {
+        this._id = _id;
+        this.email = email;
+        this.first_name = first_name;
+        this.last_name = last_name;
+        this.friends = friends;
+        this.groups = groups;
+        this.password = password;
+    }
+
+    public User(ObjectId _id, String email, String first_name, String last_name, String password) {
+        this._id = _id;
+        this.email = email;
+        this.first_name = first_name;
+        this.last_name = last_name;
+        this.friends = new ArrayList<>(0);
+        this.groups = new ArrayList<>(0);
+        this.password = password;
+    }
+
+    public User(ObjectId _id, String email, String first_name, String last_name) {
+        this._id = _id;
+        this.email = email;
+        this.first_name = first_name;
+        this.last_name = last_name;
+        this.friends = new ArrayList<>(0);
+        this.groups = new ArrayList<>(0);
+        this.password = null;
+    }
+
+    @Override
+    public boolean equals(Object o) { return o == this || o instanceof User && ((User) o)._id == _id && ((User) o).email == email; }
+
+    @Override
+    public int hashCode() {
+        return _id.hashCode();
+    }
+
+    public User load() {
+        Document user = getCollection().find(eq("_id", _id)).first();
+        Gson gson = new GsonBuilder().create();
+        return gson.fromJson(user.toJson(), User.class);
+    }
+
+    public User register(String password) throws HttpException {
+        if(this.load().equals(this)) { //need to test this, too see if it actually works!! ...
```

Comment:
Double check this works before we merge 😄 
Make sure that if it throws an Exception we catch it and throw the better HttpException instead.